### PR TITLE
feat: add a output to the command line if ignored lines are present

### PR DESCRIPTION
**Pull request by Katalam**
When merge, this PR will do:
* add an output to the command line if ignored lines are present

The idea was to display any ignored types with this new helper released today.
It was a little tricky for me to store ignored letters. Maybe you don't like the implementation, there is a lot of duplicated code. But I think perhaps another contributor would like to take his time to review the idea.

I covered only the skipped ones in gray to display them as a background color but the whole line.

## Output

CleanShot 2023-10-02 at 11 12 11

### CODE REVIEW

**Comment by devajmeireles** [comment]
Hey, @Katalam ! Thanks for your PR.

I've updated the comment adding an example of how the output will be with this PR, looks pretty good to me. I think you need to sync your branch with the current `2.x` branch because Nuno has updated the `TestCaseForTypeCoverage.php`.

File: src/TestCaseForTypeCoverage.php [hunk 1]
```diff
@@ -110,21 +111,29 @@ public function gatherAnalyserErrors(array $files): array
             foreach ($ruleRegistry->getRules($nodeType) as $rule) {
                 $ruleErrors = $rule->processNode($node, $scope);
                 foreach ($ruleErrors as $ruleError) {
-                    if (is_string($ruleError) || ! $this->ignored($ruleError)) {
-                        $actualErrors[] = $ruleErrorTransformer->transform($ruleError, $scope, $nodeType, $node->getLine());
+                    if ($this->ignored($ruleError)) {
```
Comment (devajmeireles) [inline src/TestCaseForTypeCoverage.php:119]: It looks different than the `2.x` current branch.

Comment (Katalam) [reply src/TestCaseForTypeCoverage.php:119]: I look in a minute, but I cloned it yesterday, so I'm wondered, what change since it. I will sync my fork

Comment (Katalam) [reply src/TestCaseForTypeCoverage.php:119]: Does it rly? https://github.com/pestphp/pest-plugin-type-coverage/blob/fb76bbac3b4f50f403ec0ee42c9e5301e120bfb6/src/TestCaseForTypeCoverage.php#L113-L115 this is the `2.x` branch, I can't spot a difference, I can't sync my fork, GitHub says, I'm already up-to-date

Comment (devajmeireles) [reply src/TestCaseForTypeCoverage.php:114]: > Does it rly?
> 
> https://github.com/pestphp/pest-plugin-type-coverage/blob/fb76bbac3b4f50f403ec0ee42c9e5301e120bfb6/src/TestCaseForTypeCoverage.php#L113-L115
> 
> this is the `2.x` branch, I can't spot a difference, I can't sync my fork, GitHub says, I'm already up-to-date

If it's in sync, then you've made changes. Revert and keep as original from `2.x` branch: https://github.com/pestphp/pest-plugin-type-coverage/blob/2.x/src/TestCaseForTypeCoverage.php#L93

File: src/TestCaseForTypeCoverage.php [hunk 2]
```diff
@@ -110,21 +111,29 @@ public function gatherAnalyserErrors(array $files): array
...
+        return [$actualErrors, $ignoredErrors];
     }
 
     /**
      * Check if ignored.
      */
-    private function ignored(RuleError $ruleError): bool
+    private function ignored(RuleError|string $ruleError): bool
     {
+        if (is_string($ruleError)) {
```
Comment (devajmeireles) [inline src/TestCaseForTypeCoverage.php:133]: Revert this, by preserving the `2.x` branch.

**Review by devajmeireles — APPROVED** [review]
- Reviewed
- Tested

Everything works as expected. We can go ahead with this! ✅

**Comment by devajmeireles** [comment]
@Katalam , could you fix the Static Analysis error [here](https://github.com/pestphp/pest-plugin-type-coverage/actions/runs/6372712990/job/17332120322?pr=13), please?

**Comment by Katalam** [comment]
@devajmeireles Hey, I needed to update the code a little, since it was an inherited class, but I have a good feeling about the new style

**Pull request merged**